ide/static/js/signup.js

```diff
@@ -0,0 +1,294 @@
+      error_message.innerHTML = "The password should be atleast 6 characters long.";
+      error_message.className = "error-message show";
+      password_icon.src = "static/img/password_icon_wrong.png";
+      this.setState({ error_password: 1 });
+    } else {
+      password_input.className = "input";
+      error_message.innerHTML = "";
+      error_message.className = "error-message";
+      this.setState({ error_password: 0 });
+    }
+  }
+
+  onClickConfirmPasswordInput() {
+    document.getElementById("confirm-password-input").className = "input active-input";
+    document.getElementById("confirm-password-icon").src = "static/img/password_icon_active.png";
+  }
+
+  validateSamePassword() {
+    let confirm_password = document.getElementById("confirm-password-value").value;
+    let password = document.getElementById("password-value").value;
+    let confirm_password_input = document.getElementById("confirm-password-input");
+    let confirm_password_icon = document.getElementById("confirm-password-icon");
+    let error_message = document.getElementById("error-message");
+    confirm_password_input.className = "input";
+    if (confirm_password != password) {
+      confirm_password_input.className = "input wrong-input";
+      error_message.innerHTML = "The passwords do not match.";
+      error_message.className = "error-message show";
+      confirm_password_icon.src = "static/img/password_icon_wrong.png";
+      this.setState({ error_confirm_password: 1 });
+    } else {
+      confirm_password_input.className = "input";
+      error_message.innerHTML = "";
+      error_message.className = "error-message";
+      this.setState({ error_confirm_password: 0 });
+    }
+  }
+
+  submitForm(e) {
+    e.preventDefault();
+    let username = document.getElementById("username-value").value;
+    let email = document.getElementById("email-value").value;
+    let password = document.getElementById("password-value").value;
+    password.replace(/[|&;$%@"<>()+,]/g, "");
+    let confirm_password = document.getElementById("confirm-password-value").value;
+    confirm_password.replace(/[|&;$%@"<>()+,]/g, "");
+    let cleaned_password = password.trim();
+    let cleaned_confirm_password = confirm_password.trim();
+    let csrfmiddlewaretoken = document.getElementsByName("csrfmiddlewaretoken")[0].value;
+    if (
+      this.state.error_username == 0 &&
+      this.state.error_email == 0 &&
+      this.state.error_password == 0 &&
+      this.state.error_confirm_password == 0
+    ) {
+      $.ajax({
+        method: "POST",
+        url: "/accounts/signup/",
+        data: {
+          csrfmiddlewaretoken: csrfmiddlewaretoken,
+          username: username,
+          email: email,
+          password1: cleaned_password,
+          password2: cleaned_confirm_password
+        },
+        success: function(result) {
+          if (result.location == "/") {
+            window.location = "/";
```

Comment:
@sayamkanwar we would want to set the logged in user details in user login component but we would want it to be done in a secure way can you try to figure out some method for it

Reply:
Sure @Ram81, I'll look into it.